scilpy/utils/streamlines.py

```diff
@@ -285,3 +286,48 @@ def compress_sft(sft, tol_error=0.01):
     compressed_sft.to_space(orig_space)
 
     return compressed_sft
+
+
+def cut_invalid_streamlines(sft):
+    """ Cut streamlines so their longest segment are within the bounding box """
+    sft.to_vox()
+    sft.to_corner()
+
+    copy_sft = copy.deepcopy(sft)
+    indices_to_remove, indices_to_keep = copy_sft.remove_invalid_streamlines()
+    epsilon = 0.001
+
+    new_streamlines = []
+    new_data_per_point = {}
+    for key in sft.data_per_point.keys():
+        new_data_per_point[key] = []
+
+    for ind in range(len(sft.streamlines)):
```

Comment:
instead of looking at each index why not taking all indices_to_keep and then loop just the indices_to_remove ?

Reply:
Because since I want to keep data_per_point and data_per_streamline I need to generate a new list of streamlines and two new dictionaries of data that have the same ordering as before. 

So I need the full for loop to append in the right order the cut as well as the intact streamlines